demo/src/main/java/com/smashingboxes/surelockdemo/LoginActivity.kt

```diff
@@ -0,0 +1,346 @@
+package com.smashingboxes.surelockdemo
+
+import android.animation.Animator
+import android.animation.AnimatorListenerAdapter
+import android.content.Context
+import android.content.SharedPreferences
+import android.os.AsyncTask
+import android.os.Bundle
+import android.support.v7.app.AlertDialog
+import android.support.v7.app.AppCompatActivity
+import android.text.TextUtils
+import android.util.Log
+import android.view.View
+import android.view.inputmethod.EditorInfo
+import android.widget.Button
+import android.widget.TextView
+import android.widget.Toast
+import com.smashingboxes.surelock.*
+import kotlinx.android.synthetic.main.activity_login.*
+import java.io.UnsupportedEncodingException
+
+/**
+ * Created by Tyler McCraw on 2/17/17.
+ *
+ *
+ * A login screen that offers login via username/password
+ * along with the option to set up future fingerprint authentication.
+ *
+ */
+
+class LoginActivity : AppCompatActivity(), SurelockFingerprintListener {
+
+    private var authTask: UserLoginTask? = null
+    private var surelock: Surelock? = null
+    private lateinit var surelockStorage: SurelockStorage
+    private var preferences: SharedPreferences? = null
+
+    private val preferenceHelper: SharedPreferences
+        get() {
+            if (preferences == null) {
+                preferences = getSharedPreferences(SHARED_PREFS_FILE_NAME, Context.MODE_PRIVATE)
+            }
+            return preferences!!
+        }
```

Comment:
We probably don't need two instances of SharedPreferences here.
Just make it a lazy property.
```
private val preferenceHelper: SharedPreferences by lazy {
   getSharedPreferences(SHARED_PREFS_FILE_NAME, Context.MODE_PRIVATE)
}
```